SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -10826,6 +10849,73 @@ void CEulerSolver::BC_Sym_Plane(CGeometry *geometry, CSolver **solver_container,
   
 }
 
+void CEulerSolver::BC_Fluid_Interface(CGeometry *geometry, CSolver **solver_container, CNumerics *numerics,
+                                         CConfig *config) {
+  
+  unsigned long iVertex, iPoint;
+  unsigned short iDim, iVar, iMarker;
+  
+  bool implicit 	 = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+  bool grid_movement = config->GetGrid_Movement();
+  
+  su2double Density, sq_vel, StaticEnergy;
+  
+  su2double *Normal = new su2double[nDim];
+  su2double *PrimVar_i = new su2double[nPrimVar];
+  su2double *PrimVar_j = new su2double[nPrimVar];
+  
+  int rank = MASTER_NODE, irank = 1;
+  
+    for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+
+      if (config->GetMarker_All_KindBC(iMarker) == FLUID_INTERFACE)	{
+
+        for (iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+          iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+
+          if (geometry->node[iPoint]->GetDomain()) {
+
```

Comment:
I think here you could add the calculation for the secondary variables so to allow real gas calculation (it would work just for air fluid model now)